imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/TIFFImageWriter.java

```diff
@@ -632,127 +637,126 @@ private void writeImageData(DataOutput stream, RenderedImage renderedImage, int
         final int tileWidth = renderedImage.getTileWidth();
 
         // TODO: SampleSize may differ between bands/banks
-        int sampleSize = renderedImage.getSampleModel().getSampleSize(0);
+        SampleModel sampleModel = renderedImage.getSampleModel();
+
+        int[] sampleSizes = sampleModel.getSampleSize();
+        int pixelSize = 0;
+        int sampleSize = sampleSizes[0];
+        for (int i = 0; i < sampleSizes.length; i++) {
+            pixelSize += sampleSize;
+            if (sampleSize != sampleSizes[i]) {
+                throw new IIOException("Variable BitsPerSample not supported: " + Arrays.toString(sampleSizes));
+            }
+        }
+
         final ByteBuffer buffer;
-        if (sampleSize == 1) {
+        if (pixelSize == 1) {
             buffer = ByteBuffer.allocate((tileWidth + 7) / 8);
         }
         else {
-            buffer = ByteBuffer.allocate(tileWidth * renderedImage.getSampleModel().getNumBands() * sampleSize / 8);
+            buffer = ByteBuffer.allocate(tileWidth * pixelSize / 8);
```

Comment:
You are right. Also this definately causes problems with pixelsizes that are not full bytes and greater that 1 bit.